pkgs/applications/networking/misc/zammad/default.nix

```diff
@@ -149,9 +137,11 @@ stdenv.mkDerivation {
 
   installPhase = ''
     cp -R . $out
-    cp ${databaseConfig} $out/config/database.yml
-    cp ${secretsConfig} $out/config/secrets.yml
-    sed -i -e "s|info|debug|" $out/config/environments/production.rb
+    rm -rf $out/config/database.yml $out/config/secrets.yml $out/tmp $out/log
+    ln -s ${dataDir}/config/database.yml $out/config/database.yml
+    ln -s ${dataDir}/config/secrets.yml $out/config/secrets.yml
+    ln -s ${dataDir}/tmp $out/tmp
+    ln -s ${dataDir}/log $out/log
```

Comment:
If I understand correctly, we are symlinking these so that *at runtime* they point to the *dataDir*. They don't have to exist at build time at all.

I think it's worth putting a comment explaining what's going on. It took me a few attempts to figure it out.

Reply:
I have added a comment there to make this more clear.

Reply:
This symlinks the data directory into the Nix store, surely there must be another way?

Reply:
This is normally also the way to handle the `dataDir` for big PHP-Packages. Just grep for `dataDir` in the pkgs and you will find many packages where this symlink-hack is used to rewrite the `dataDir` to its actual location.

Reply:
I see, thanks for the explanation!